src/components/common/formFields/Widgets.tsx

```diff
@@ -0,0 +1,366 @@
+import './Widgets.scss'
+
+interface WidgetPropsType {
+    title: string
+    description?: string
+    rootClassName?: string
+    onChange?: (e) => void
+    onBlur?: (e) => void
+    minLength?: number
+    maxLength?: number
+    isRequired?: boolean
+    placeholder?: string
+    errorMessage?: string
+}
+
+const renderValidationErrorLabel = (message: string): JSX.Element => {
+    return (
+        <div className="error-label flex left align-start fs-11 fw-4 mt-6">
+            <span className="error-label-icon">
+                <Error className="icon-dim-16" />
+            </span>
+            <span className="ml-4 cr-5">{message}</span>
+        </div>
+    )
+}
+
+const renderDescription = (description: string) => {
+    return (
+        <div className="info-label flex left align-center fs-11 fw-4 mt-4">
+            <span className="info-label-icon">
+                <Info className="icon-dim-16" />
+            </span>
+            <span className="ml-4 cn-7">{description}</span>
+        </div>
+    )
+}
+
+interface StyledFieldPropsType {
+    title: string
+    description?: string
+    rootClassName?: string
+    isRequired?: boolean
+    errorMessage?: string
+    showBox?: boolean
+    children: React.ReactNode
+}
+
+const StyledField = ({
+    title,
+    description,
+    rootClassName,
+    isRequired,
+    errorMessage,
+    children,
+}: StyledFieldPropsType) => {
+    return (
+        <label className={`styled-field form__row form__row--w-100 ${rootClassName || ''}`}>
+            <span className={`form__label fs-13 cn-9 mb-6${isRequired ? ' required-field' : ''}`}>{title}</span>
+            {children}
+            {description && renderDescription(description)}
+            {errorMessage && renderValidationErrorLabel(errorMessage)}
+        </label>
+    )
+}
+
+export const StyledFormBox = ({ title, description, showBox, children }: StyledFieldPropsType) => {
+    return (
+        <div className={`styled-form-box ${showBox ? 'show-box' : ''}`}>
+            <h2 className="fs-14 fw-6 p-0 m-0 lh-20">{title}</h2>
+            {description && <p className="fs-11 fw-4 p-0 mt-4 lh-16">{description}</p>}
+            <hr />
+            {children}
+        </div>
+    )
+}
+
+interface StyledInputPropsType extends WidgetPropsType {
+    value: string
+    pattern?: string
+}
+
+export const StyledInput = ({
+    title,
+    value,
+    description,
+    rootClassName,
+    onChange,
+    onBlur,
+    minLength,
+    maxLength,
+    pattern,
+    isRequired,
+    placeholder,
+    errorMessage,
+}: StyledInputPropsType): JSX.Element => {
+    return (
+        <StyledField
+            title={title}
+            description={description}
+            rootClassName={rootClassName}
+            isRequired={isRequired}
+            errorMessage={errorMessage}
+        >
+            <input
+                placeholder={placeholder && placeholder}
+                name={title.replace(/\s/g, '').toLowerCase()}
```

Comment:
Can we replace space with _ or something for better readability?

Reply:
Done